runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -262,10 +263,18 @@ class ExecutionContext final {
     std::unique_ptr<MemoryManager> memory_man_ptr{nullptr};
     std::unique_ptr<PythonInterpreterGuard> py_guard{nullptr};
 
+    // PRNG
+    std::string seed;
+    std::mt19937 gen;
+
   public:
-    explicit ExecutionContext() : initial_tape_recorder_status(false)
+    explicit ExecutionContext(std::string _seed = "") : initial_tape_recorder_status(false)
```

Comment:
It's just me being extra pedantic. It is unnecessary.

Reply:
Just asking because I would like to know :) In Python we use `_*` to indicate a variable is unused, but that doesn't appear to be the case here.

Reply:
Since the class attribute is also named `gen` I was afraid of using `gen` as the setter argument since it would lead to things like `gen = _gen`, or here `seed = seed`. But I guess `func(argtype seed) {this->seed = seed;}` is perfectly fine and unambiguous.  :D

It isn't really a convention though, just my (bad?) habit.

Reply:
These underscores are now removed.

Reply:
Ah got it, thanks :) yeah `this` is what I've seen to disambiguate